docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversion/MapKeyToVector.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic
+{
+    class MapKeyToVector
+    {
+        /// This example demonstrates the use of MapKeyToVector by mapping keys to floats[]. 
+        /// Because the ML.NET KeyType maps the missing value to zero, counting of uints starts at 1, so the values
+        /// converted to KeyTypes will appear skewed by one. See  https://github.com/dotnet/machinelearning/issues/3072
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable.
+            var rawData = new[] {
+                new DataPoint() { Timeframe = 9, Category = 5 },
+                new DataPoint() { Timeframe = 8, Category = 4 },
+                new DataPoint() { Timeframe = 8, Category = 4 },
+                new DataPoint() { Timeframe = 9, Category = 3 },
+                new DataPoint() { Timeframe = 2, Category = 3 },
+                new DataPoint() { Timeframe = 3, Category = 5 }
+            };
+
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            // Constructs the ML.net pipeline
+            var pipeline = mlContext.Transforms.Conversion.MapKeyToVector("TimeframeVector", "Timeframe")
+                           .Append(mlContext.Transforms.Conversion.MapKeyToVector("CategoryVector", "Category", outputCountVector: true));
+
+            // Fits the pipeline to the data.
+            IDataView transformedData = pipeline.Fit(data).Transform(data);
+
+            // Getting the resulting data as an IEnumerable.
+            // This will contain the newly created columns.
+            IEnumerable<TransformedData> features = mlContext.Data.CreateEnumerable<TransformedData>(transformedData, reuseRowObject: false);
+
+            Console.WriteLine($" Timeframe           TimeframeVector         Category    CategoryVector");
+            foreach (var featureRow in features)
+                Console.WriteLine($"{featureRow.Timeframe}\t\t\t{string.Join(',', featureRow.TimeframeVector)}\t\t\t{featureRow.Category}\t\t{string.Join(',', featureRow.CategoryVector)}");
+
+            // TransformedData obtained post-transformation.
+            //
+            // Timeframe          TimeframeVector    Category    CategoryVector
+            //  10              0,0,0,0,0,0,0,0,0,1       6          0,0,0,0,0
```

Comment:
Is this correct? The first `Timeframe` was `9` in your declaration. If no transformer is applied to `Timeframe`, ML.NET should not touch its value? Is my understanding correct? Or I am missing something?

Reply:
Yeah.... this is why i logged 3072, and had a discussion on Friday post scrum. Gets easier to think of them as strings, not numbers. 

the string "0", gets mapped to the first available ulong, 1. (The ulong 0 is reserved for missing key).
The string "1" gets mapped to the next available : 2. 
On our KeyType system counting starts from 1. 0 is reserved. 

---
In reply to: [272774353](https://github.com/dotnet/machinelearning/pull/3211#discussion_r272774353) [](ancestors = 272774353)